### ISSUE TITLE
Picker width on Windows not filling container

### ISSUE THREAD

**Issue by mike3sullivan** [dotnet/maui#6391]
### Description

When placing a Picker control inside a grid, it does not resize to fill the grid cell on Windows like it does on Android.

### Steps to Reproduce

[Maui Picker Width.zip](https://github.com/dotnet/maui/files/8535278/Maui.Picker.Width.zip)
Place a picker control inside a grid -- added backgroundcolors and a margin to show relative sizes.
On Android, it shows the Picker fills the cell; on Windows it shows the Picker as a minimal size inside the larger(blue) cell.
`
    <Grid ColumnDefinitions="*,*,*" RowDefinitions="*,*,*"  >
        <Grid Grid.Column="1" Grid.Row="1"  BackgroundColor="Blue">
        <Picker SelectedItem="Baboon" BackgroundColor="Green" Margin="3">
            <Picker.Items>
                <x:String>Baboon</x:String>
                <x:String>Capuchin Monkey</x:String>
            </Picker.Items>
        </Picker>
        </Grid>
    </Grid>`

![picker size android](https://user-images.githubusercontent.com/26934129/164557817-2cbb262e-0d91-41a2-a802-f127f6bcdfeb.png)
![picker size](https://user-images.githubusercontent.com/26934129/164557835-a9e7453b-277c-43c7-948e-dfc7c5c7a693.png)

### Version with bug

Release Candidate 1 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows 10

### Did you find any workaround?

setting width and height manually

**Comment by XamlTest** [comment]
Verified this issue with Visual Studio Enterprise 17.3.0 Preview 1.0 [32414.199.main]. Repro on Windows with above project.

**Comment by kristinx0211** [comment]
still an issue on windows with 17.3.0 Preview 2.0 [32507.29.main].

**Comment by austinthale** [comment]
On top of that, when a title is set, it appears above the actual picker area instead of inside of it:
![image](https://user-images.githubusercontent.com/25466210/171983668-746f08d2-928f-499b-847e-91c27c13b63e.png)

**Comment by notanaverageman** [comment]
To workaround this issue I have mapped the `HorizontalOptions` of `Picker` to `HorizontalAlignment` of `ComboBox` on Windows.

```csharp

#if WINDOWS
...
PickerHandler.Mapper.Add(nameof(View.HorizontalOptions), MapHorizontalOptions);
...

private static void MapHorizontalOptions(IViewHandler handler, IView view)
{
	if (view is not View mauiView)
	{
		return;
	}

	if (handler.PlatformView is not Microsoft.UI.Xaml.FrameworkElement element)
	{
		return;
	}

	element.HorizontalAlignment = mauiView.HorizontalOptions.Alignment switch
	{
		LayoutAlignment.Start  => Microsoft.UI.Xaml.HorizontalAlignment.Left,
		LayoutAlignment.Center => Microsoft.UI.Xaml.HorizontalAlignment.Center,
		LayoutAlignment.End    => Microsoft.UI.Xaml.HorizontalAlignment.Right,
		LayoutAlignment.Fill   => Microsoft.UI.Xaml.HorizontalAlignment.Stretch,
		_ => throw new ArgumentOutOfRangeException()
	};
}
#endif

```

**Comment by nk54** [comment]
I hate this picker look on Windows... If you set a width, nice it's ok. Select an item, the width of the picker is updated to be the size of the selected item. If I set my Picker width to be 200, don't change it to 100 or 120 or 150 depends on the length of the selected item display string value...

**Comment by davepruitt** [comment]
Currently experiencing this bug as well.

**Comment by Mzazvor** [comment]
Im still having this issue with 17.3.4.

**Comment by NPadrutt** [comment]
yeah, I still experience this issue as well.. Given it is in the backlog for more than half a year at this point it seems that doesn't change anytime soon.

**Comment by williambohrmann3** [comment]
Also able to repro

**Comment by returnNnull** [comment]
Same problem

**Comment by nicop85** [comment]
+1 to upvote this issue in the priority list
